src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -70,12 +70,53 @@ public String greeting(@RequestParam(name = "name", required = false, defaultVal
         return bookRepository.findByTitle(bookTitle);
     }
 
-    @GetMapping("/publisherGenreYear/{publisher}/{genre}/{year}")
-    public List<Book> findByGenrePublisherYear(@PathVariable String publisher, @PathVariable String genre,
-            @PathVariable String year) {
+    @GetMapping("/publisherGenreYear")
+    public List<Book> findByGenrePublisherYear(@RequestParam(required = false) String publisher,
+            @RequestParam(required = false) String genre,
+            @RequestParam(required = false) String year) {
         return bookRepository.findByPublisherAndGenreAndYear(publisher, genre, year);
     }
 
+    @GetMapping("/allBooks")
+    public List<Book> findByAnyParam(@RequestParam(required = false, defaultValue = "") String publisher,
+            @RequestParam(required = false, defaultValue = "") String genre,
```

Comment:
If you need all values as null by default you should remove the 'defaultValue= ""' so you won't need to do all those if below

Reply:
fixed at last commit